src/OrleansRuntime/Core/InsideRuntimeClient.cs

```diff
@@ -357,11 +373,13 @@ private Task<object> InvokeWithInterceptors(IAddressable target, InvokeMethodReq
         {
             // If the target has a grain-level interceptor or there is a silo-level interceptor, intercept the
             // call.
-            var siloWideInterceptor = this.CurrentStreamProviderRuntime.GetInvokeInterceptor();
             var grainWithInterceptor = target as IGrainInvokeInterceptor;
 
             // Silo-wide interceptors do not operate on system targets.
-            var hasSiloWideInterceptor = siloWideInterceptor != null && target is IGrain;
+#pragma warning disable 618
+            var hasSiloWideInterceptor = target is IGrain && (this.CurrentStreamProviderRuntime?.GetInvokeInterceptor() != null ||
+#pragma warning restore 618
+                                          this.siloInterceptors.Count > 1);
```

Comment:
It's quite unfortunate. How would it look if I treated it separately? I'm not sure